In [3]:
from datascience import *
import pandas as pd
import seaborn as sns
import numpy as np
import math

from astropy import units as u
from astropy.time import Time
from astropy.coordinates import SkyCoord

%matplotlib inline
import matplotlib.pyplot as plots
plots.style.use('fivethirtyeight')

# Anexo: Estrellas Prominentes del Catálogo de Hipparcos
### https://es.wikipedia.org/wiki/Anexo:Estrellas_del_Cat%C3%A1logo_Hipparcos
### https://www.cosmos.esa.int/web/hipparcos/sample-tables-1

In [4]:
hip_main = Table.read_table("hip_cat_2.csv", sep=',')
hip_main.show(2)

In [5]:
def pcToLy(pc):
    ly = pc * 3.26
    return ly

pcToLy(61.1247)

199.26652199999998

In [6]:
def lyToPc(ly):
    pc = ly / 3.26
    return pc

lyToPc(97.023794)

29.7619

In [7]:
lyToPc(97)

29.754601226993866

In [8]:
# HIP 1 ~ 10000
dic = {'HIP': [667], 'Estrella': ['α Andromedae (Alpheratz o Sirrah)'], 'dist (pc)': [29.7619], 'dist (ANEXO)': [lyToPc(97)]} 

dic['HIP'].append(3419)
dic['Estrella'].append('β Ceti (Deneb Kaitos o Diphda)')
dic['HIP'].append(3829)
dic['Estrella'].append('Estrella de Van Maanen')    
dic['HIP'].append(5447)
dic['Estrella'].append('β Andromedae (Mirach)')
dic['HIP'].append(7588)
dic['Estrella'].append('α Eridani (Achernar)')
dic['HIP'].append(9884)
dic['Estrella'].append('α Arietis (Hamal)')

# HIP 10001 ~ 20000
dic['HIP'].append(10826)
dic['Estrella'].append('ο Ceti (Mira)')
dic['HIP'].append(11767)
dic['Estrella'].append('α Ursae Minoris (Polaris)')
dic['HIP'].append(15863)
dic['Estrella'].append('α Persei (Mirfak o Algenib)')

# HIP 20001 ~ 30000
dic['HIP'].append(21421)
dic['Estrella'].append('α Tauri (Aldebarán)')
dic['HIP'].append(24186)
dic['Estrella'].append('Estrella de Kapteyn')
dic['HIP'].append(24436)
dic['Estrella'].append('β Orionis (Rigel)')
dic['HIP'].append(24608)
dic['Estrella'].append('α Aurigae (Capella)')
dic['HIP'].append(25336)
dic['Estrella'].append('γ Orionis (Bellatrix)')
dic['HIP'].append(25428)
dic['Estrella'].append('β Tauri (Elnath)')
dic['HIP'].append(25930)
dic['Estrella'].append('δ Orionis (Mintaka)')
dic['HIP'].append(26311)
dic['Estrella'].append('ε Orionis (Alnilam)')
dic['HIP'].append(26727)
dic['Estrella'].append('ζ Orionis (Alnitak)')
dic['HIP'].append(27366)
dic['Estrella'].append('κ Orionis (Saiph)')
dic['HIP'].append(27989)
dic['Estrella'].append('α Orionis (Betelgeuse)')
dic['HIP'].append(28360)
dic['Estrella'].append('β Aurigae (Menkalinan)')

# HIP 30001 ~ 40000
dic['HIP'].append(30324)
dic['Estrella'].append('β Canis Majoris (Mirzam o Murzim)')
dic['HIP'].append(30438)
dic['Estrella'].append('α Carinae (Canopo)')
dic['HIP'].append(31681)
dic['Estrella'].append('γ Geminorum (Alhena)')
dic['HIP'].append(32349)
dic['Estrella'].append('α Canis Majoris (Sirio)')
dic['HIP'].append(33579)
dic['Estrella'].append('ε Canis Majoris (Adhara)')
dic['HIP'].append(34444)
dic['Estrella'].append('δ Canis Majoris (Wezen)')
dic['HIP'].append(36850)
dic['Estrella'].append('α Geminorum (Cástor)')
dic['HIP'].append(37279)
dic['Estrella'].append('α Canis Minoris (Procyon)')
dic['HIP'].append(37826)
dic['Estrella'].append('β Geminorum (Pólux)')
dic['HIP'].append(39953)
dic['Estrella'].append('γ Velorum (Suhail o Regor)')

# HIP 40001 ~ 50000
dic['HIP'].append(41037)
dic['Estrella'].append('ε Carinae (Avior)')
dic['HIP'].append(42913)
dic['Estrella'].append('δ Velorum')
dic['HIP'].append(45238)
dic['Estrella'].append('β Carinae (Miaplacidus)')
dic['HIP'].append(46390)
dic['Estrella'].append('α Hydrae (Alphard)')
dic['HIP'].append(49669)
dic['Estrella'].append('α Leonis (Regulus)')        

# HIP 50001 ~ 60000
dic['HIP'].append(50583)
dic['Estrella'].append('γ Leonis (Algieba)')
dic['HIP'].append(54035)
dic['Estrella'].append('Lalande 21185')
dic['HIP'].append(54061)
dic['Estrella'].append('α Ursae Majoris (Dubhe)')
dic['HIP'].append(54872)
dic['Estrella'].append('δ Leonis (Duhr o Zosma)')
dic['HIP'].append(54879)
dic['Estrella'].append('θ Leonis (Chertan)')
dic['HIP'].append(56997)
dic['Estrella'].append('61 Ursae Majoris')
dic['HIP'].append(57362)
dic['Estrella'].append('β Leonis (Denébola)')
dic['HIP'].append(57757)
dic['Estrella'].append('β Virginis (Zavijava)')
dic['HIP'].append(58001)
dic['Estrella'].append('γ Ursae Majoris (Phecda)')
dic['HIP'].append(59196)
dic['Estrella'].append('δ Centauri')

# HIP 60001 ~ 70000
dic['HIP'].append(60351)
dic['Estrella'].append('12 Comae Berenices')
dic['HIP'].append(60718)
dic['Estrella'].append('α Crucis (Acrux)')
dic['HIP'].append(61084)
dic['Estrella'].append('γ Crucis (Gacrux)')
dic['HIP'].append(62434)
dic['Estrella'].append('β Crucis (Becrux o Mimosa)')
dic['HIP'].append(62956)
dic['Estrella'].append('ε Ursae Majoris (Alioth)')
dic['HIP'].append(63608)
dic['Estrella'].append('ε Virginis (Vindemiatrix)')
dic['HIP'].append(65474)
dic['Estrella'].append('α Virginis (Espiga)')
dic['HIP'].append(65835)
dic['Estrella'].append('R Hydrae')
dic['HIP'].append(66249)
dic['Estrella'].append('ζ Virginis (Heze)')
dic['HIP'].append(67301)
dic['Estrella'].append('η Ursae Majoris (Alkaid o Benetnasch)')
dic['HIP'].append(68702)
dic['Estrella'].append('β Centauri (Hadar o Agena)')
dic['HIP'].append(68933)
dic['Estrella'].append('θ Centauri (Menkent)')
dic['HIP'].append(69673)
dic['Estrella'].append('α Bootis (Arturo)')

# HIP 70001 ~ 80000
dic['HIP'].append(71681)
dic['Estrella'].append('α Centauri B (Rigil Kentaurus o Toliman)')
dic['HIP'].append(71683)
dic['Estrella'].append('α Centauri A (Rigil Kentaurus o Toliman)')
dic['HIP'].append(71860)
dic['Estrella'].append('α Lupi (Kakkab o Men)')
dic['HIP'].append(72607)
dic['Estrella'].append('β Ursae Minoris (Kochab)')
dic['HIP'].append(75097)
dic['Estrella'].append('γ Ursae Minoris (Pherkad)')

# HIP 80001 ~ 90000
dic['HIP'].append(80763)
dic['Estrella'].append('α Scorpii (Antares)')
dic['HIP'].append(82273)
dic['Estrella'].append('α Trianguli Australis (Atria)')
dic['HIP'].append(82396)
dic['Estrella'].append('ε Scorpii (Wei)')
dic['HIP'].append(85927)
dic['Estrella'].append('λ Scorpii (Shaula)')
dic['HIP'].append(86032)
dic['Estrella'].append('α Ophiuchi (Rasalhague o Ras Alhague)')
dic['HIP'].append(86228)
dic['Estrella'].append('θ Scorpii (Sargas o Girtab)')

# HIP 90001 ~ 100000
dic['HIP'].append(90185)
dic['Estrella'].append('ε Sagittarii (Kaus Australis)')
dic['HIP'].append(90496)
dic['Estrella'].append('λ Sagittarii (Kaus Borealis)')
dic['HIP'].append(91262)
dic['Estrella'].append('α Lyrae (Vega)')
dic['HIP'].append(92403)
dic['Estrella'].append('Ross 154')
dic['HIP'].append(92855)
dic['Estrella'].append('σ Sagittarii (Nunki)')
dic['HIP'].append(97278)
dic['Estrella'].append('γ Aquilae (Tarazed)')
dic['HIP'].append(97649)
dic['Estrella'].append('α Aquilae (Altair)')
        
# HIP 100001 ~ 110000
dic['HIP'].append(100751)
dic['Estrella'].append('α Pavonis (Peacock)')
dic['HIP'].append(102098)
dic['Estrella'].append('α Cygni (Deneb, Arided o Aridif)')
dic['HIP'].append(105881)
dic['Estrella'].append('ζ Capricorni (Marakk)')
dic['HIP'].append(109268)
dic['Estrella'].append('α Gruis (Alnair)')
dic['HIP'].append(109358)
dic['Estrella'].append('β Canum Venaticorum (Asterion o Chara)')

# HIP 110001 ~ 120000
dic['HIP'].append(112122)
dic['Estrella'].append('β Gruis (Gruid)')
dic['HIP'].append(113368)
dic['Estrella'].append('α Piscis Austrini (Fomalhaut)')
        

In [9]:
len(dic['HIP'])

84

In [10]:
len(dic['Estrella'])

84

In [11]:
len(dic['dist (pc)'])

1

In [12]:
len(dic['dist (ANEXO)'])

1

In [13]:
a = Table().with_columns(
    'Estrellas', dic['Estrella'],
    'HIP', dic['HIP'],
)
a

Estrellas                         | HIP
α Andromedae (Alpheratz o Sirrah) | 667
β Ceti (Deneb Kaitos o Diphda)    | 3419
Estrella de Van Maanen            | 3829
β Andromedae (Mirach)             | 5447
α Eridani (Achernar)              | 7588
α Arietis (Hamal)                 | 9884
ο Ceti (Mira)                     | 10826
α Ursae Minoris (Polaris)         | 11767
α Persei (Mirfak o Algenib)       | 15863
α Tauri (Aldebarán)               | 21421
... (74 rows omitted)

In [14]:
estrellasCatalogoHipparcos = hip_main.select(0, 6)
e_cat_hip = estrellasCatalogoHipparcos
e_cat_hip

HIP  | dist (pc)
1    | 282.486
2    | 45.6621
3    | 355.872
4    | 129.032
5    | 348.432
6    | 53.1915
7    | 56.3698
8    | 193.424
9    | 207.9
10   | 92.9368
... (113700 rows omitted)

In [15]:
for i in range(1, len(dic['HIP'])):
    dic['dist (pc)'].append(e_cat_hip.where(0, are.equal_to(dic['HIP'][i])).column(1)[0])

len(dic['dist (pc)'])

84

In [16]:
dic['dist (ANEXO)'].append(lyToPc(96))
dic['dist (ANEXO)'].append(lyToPc(14.1))
dic['dist (ANEXO)'].append(lyToPc(199))
dic['dist (ANEXO)'].append(lyToPc(144))
dic['dist (ANEXO)'].append(lyToPc(66))
dic['dist (ANEXO)'].append(lyToPc(418))
dic['dist (ANEXO)'].append(lyToPc(431))
dic['dist (ANEXO)'].append(lyToPc(590))
dic['dist (ANEXO)'].append(lyToPc(65.1))
dic['dist (ANEXO)'].append(lyToPc(12.8))
dic['dist (ANEXO)'].append(lyToPc(860))
dic['dist (ANEXO)'].append(lyToPc(42.2))
dic['dist (ANEXO)'].append(lyToPc(240))
dic['dist (ANEXO)'].append(lyToPc(131))
dic['dist (ANEXO)'].append(lyToPc(915))
dic['dist (ANEXO)'].append(lyToPc(1340))
dic['dist (ANEXO)'].append(lyToPc(700))
dic['dist (ANEXO)'].append(lyToPc(722))
dic['dist (ANEXO)'].append(lyToPc(643))
dic['dist (ANEXO)'].append(lyToPc(82.1))
dic['dist (ANEXO)'].append(lyToPc(500))
dic['dist (ANEXO)'].append(lyToPc(309))
dic['dist (ANEXO)'].append(lyToPc(105))
dic['dist (ANEXO)'].append(lyToPc(8.6))
dic['dist (ANEXO)'].append(lyToPc(405))
dic['dist (ANEXO)'].append(lyToPc(1800))
dic['dist (ANEXO)'].append(lyToPc(51))
dic['dist (ANEXO)'].append(lyToPc(11.41))
dic['dist (ANEXO)'].append(lyToPc(33.7))
dic['dist (ANEXO)'].append(lyToPc(800))
dic['dist (ANEXO)'].append(lyToPc(630))
dic['dist (ANEXO)'].append(lyToPc(79.7))
dic['dist (ANEXO)'].append(lyToPc(111 ))
dic['dist (ANEXO)'].append(lyToPc(177))
dic['dist (ANEXO)'].append(lyToPc(77))
dic['dist (ANEXO)'].append(lyToPc(131))
dic['dist (ANEXO)'].append(lyToPc(8.31))
dic['dist (ANEXO)'].append(lyToPc(124))
dic['dist (ANEXO)'].append(lyToPc(57.7))
dic['dist (ANEXO)'].append(lyToPc(165))
dic['dist (ANEXO)'].append(lyToPc(31.1))
dic['dist (ANEXO)'].append(lyToPc(35.9))
dic['dist (ANEXO)'].append(lyToPc(35.6))
dic['dist (ANEXO)'].append(lyToPc(84))
dic['dist (ANEXO)'].append(lyToPc(395))
dic['dist (ANEXO)'].append(lyToPc(273.394))
dic['dist (ANEXO)'].append(lyToPc(325))
dic['dist (ANEXO)'].append(lyToPc(88))
dic['dist (ANEXO)'].append(lyToPc(280))
dic['dist (ANEXO)'].append(lyToPc(81))
dic['dist (ANEXO)'].append(lyToPc(110))
dic['dist (ANEXO)'].append(lyToPc(260))
dic['dist (ANEXO)'].append(lyToPc(425))
dic['dist (ANEXO)'].append(lyToPc(73.2))
dic['dist (ANEXO)'].append(lyToPc(101))
dic['dist (ANEXO)'].append(lyToPc(352))
dic['dist (ANEXO)'].append(lyToPc(60.9))
dic['dist (ANEXO)'].append(lyToPc(36.7))
dic['dist (ANEXO)'].append(lyToPc(4.5))
dic['dist (ANEXO)'].append(lyToPc(4.5))
dic['dist (ANEXO)'].append(lyToPc(550))
dic['dist (ANEXO)'].append(lyToPc(129))
dic['dist (ANEXO)'].append(lyToPc(480))
dic['dist (ANEXO)'].append(lyToPc(550))
dic['dist (ANEXO)'].append(lyToPc(415))
dic['dist (ANEXO)'].append(lyToPc(65))
dic['dist (ANEXO)'].append(lyToPc(570))
dic['dist (ANEXO)'].append(lyToPc(46.7))
dic['dist (ANEXO)'].append(lyToPc(270))
dic['dist (ANEXO)'].append(lyToPc(145))
dic['dist (ANEXO)'].append(lyToPc(77.3))
dic['dist (ANEXO)'].append(lyToPc(25.3))
dic['dist (ANEXO)'].append(lyToPc(9.68))
dic['dist (ANEXO)'].append(lyToPc(224))
dic['dist (ANEXO)'].append(lyToPc(461))
dic['dist (ANEXO)'].append(lyToPc(16.73))
dic['dist (ANEXO)'].append(lyToPc(183))
dic['dist (ANEXO)'].append(lyToPc(1425))
dic['dist (ANEXO)'].append(lyToPc(398))
dic['dist (ANEXO)'].append(lyToPc(101))
dic['dist (ANEXO)'].append(lyToPc(27.4))
dic['dist (ANEXO)'].append(lyToPc(170))
dic['dist (ANEXO)'].append(lyToPc(25))

In [17]:
len(dic['dist (ANEXO)'])
#dic['dist (ANEXO)']

84

In [18]:
e = a.with_columns(
    'dist (Eric)', dic['dist (pc)'],
    'dist (ANEXO)', dic['dist (ANEXO)']
)
e

Estrellas                         | HIP   | dist (Eric) | dist (ANEXO)
α Andromedae (Alpheratz o Sirrah) | 667   | 29.7619     | 29.7546
β Ceti (Deneb Kaitos o Diphda)    | 3419  | 29.3772     | 29.4479
Estrella de Van Maanen            | 3829  | 4.40626     | 4.32515
β Andromedae (Mirach)             | 5447  | 61.1247     | 61.0429
α Eridani (Achernar)              | 7588  | 44.0917     | 44.1718
α Arietis (Hamal)                 | 9884  | 20.2102     | 20.2454
ο Ceti (Mira)                     | 10826 | 128.37      | 128.221
α Ursae Minoris (Polaris)         | 11767 | 132.275     | 132.209
α Persei (Mirfak o Algenib)       | 15863 | 181.488     | 180.982
α Tauri (Aldebarán)               | 21421 | 19.9641     | 19.9693
... (74 rows omitted)

In [19]:
#e.to_csv('AnexoEstrellasProminentes.csv')

In [20]:
tb = Table.read_table('AnexoEstrellasProminentes.csv')
tb.show()

FileNotFoundError: [Errno 2] File AnexoEstrellasProminentes.csv does not exist: 'AnexoEstrellasProminentes.csv'

In [ ]:
print(lyToPc(1425))
pcToLy(990.099)

In [ ]:
this_dict = {'Vmag' : [2.07], 'AbsMag': [-0.298304], 'AbsMag_Anexo': ['-0,30']}

this_dict

In [ ]:
for i in range(1, len(dic['HIP'])):
    this_dict['Vmag'].append(hip_main.where(0, are.equal_to(dic['HIP'][i])).column(3)[0])

In [ ]:
#del this_dict['Vmag']

In [ ]:
for i in range(1, len(dic['HIP'])):
    this_dict['AbsMag'].append(hip_main.where(0, are.equal_to(dic['HIP'][i])).column(5)[0])

In [ ]:
#del this_dict['AbsMag']

In [ ]:
len(this_dict['Vmag'])
#this_dict['AbsMag']

In [ ]:
this_dict['AbsMag_Anexo'].append('-0,31')
this_dict['AbsMag_Anexo'].append('+14,2')
this_dict['AbsMag_Anexo'].append('-1,87')
this_dict['AbsMag_Anexo'].append('-2.77')
this_dict['AbsMag_Anexo'].append('+0,48')
this_dict['AbsMag_Anexo'].append('+0,93')
this_dict['AbsMag_Anexo'].append('-3,65')
this_dict['AbsMag_Anexo'].append('-4,50')
this_dict['AbsMag_Anexo'].append('−0,63 / +11,98')
this_dict['AbsMag_Anexo'].append('+10,88')
this_dict['AbsMag_Anexo'].append('-6,7  ')
this_dict['AbsMag_Anexo'].append('-0,5 (+0,14 / +0,29)')
this_dict['AbsMag_Anexo'].append('-2,72')
this_dict['AbsMag_Anexo'].append('-1,34')
this_dict['AbsMag_Anexo'].append('-4,99 (conjunta)')
this_dict['AbsMag_Anexo'].append('-6.95')
this_dict['AbsMag_Anexo'].append('-5,10')
this_dict['AbsMag_Anexo'].append('-4,68')
this_dict['AbsMag_Anexo'].append('-6,0 a -5,5')
this_dict['AbsMag_Anexo'].append('-0,11 (conjunta)')
this_dict['AbsMag_Anexo'].append('-3,95')
this_dict['AbsMag_Anexo'].append('-5,53')
this_dict['AbsMag_Anexo'].append('-0,61 ')
this_dict['AbsMag_Anexo'].append('1,42 (A) / 11,18 (B)')
this_dict['AbsMag_Anexo'].append('-4,10')
this_dict['AbsMag_Anexo'].append('-6,87')
this_dict['AbsMag_Anexo'].append('+1,22 / +2,8')
this_dict['AbsMag_Anexo'].append('+2,65 / +13,04')
this_dict['AbsMag_Anexo'].append('+1,09')
this_dict['AbsMag_Anexo'].append('-5,31 / -2,79')
this_dict['AbsMag_Anexo'].append('-4,58 (conjunta)')
this_dict['AbsMag_Anexo'].append('0,4 / 1,5 / 3,6')
this_dict['AbsMag_Anexo'].append('-1,00')
this_dict['AbsMag_Anexo'].append('-1,69')
this_dict['AbsMag_Anexo'].append('-0,5 / +4,2 / +9,5')
this_dict['AbsMag_Anexo'].append('-0,92 (conjunta)')
this_dict['AbsMag_Anexo'].append('+10,46')
this_dict['AbsMag_Anexo'].append('-1,09 / +1,83 / +4,22')
this_dict['AbsMag_Anexo'].append('+1,29')
this_dict['AbsMag_Anexo'].append('-0,35')
this_dict['AbsMag_Anexo'].append('+5,41')
this_dict['AbsMag_Anexo'].append('+1,91')
this_dict['AbsMag_Anexo'].append('+3,40')
this_dict['AbsMag_Anexo'].append('+0,36')
this_dict['AbsMag_Anexo'].append('-2,82')
this_dict['AbsMag_Anexo'].append('0,16 ')
this_dict['AbsMag_Anexo'].append('-3,6 / -2,88')
this_dict['AbsMag_Anexo'].append('-0,56 ')
this_dict['AbsMag_Anexo'].append('-3,92')
this_dict['AbsMag_Anexo'].append('-0,22')
this_dict['AbsMag_Anexo'].append('+0,22 ')
this_dict['AbsMag_Anexo'].append('-3,55 (conjunta)')
this_dict['AbsMag_Anexo'].append('-2,55')
this_dict['AbsMag_Anexo'].append('+1,62')
this_dict['AbsMag_Anexo'].append('-0,60')
this_dict['AbsMag_Anexo'].append('-5,42')
this_dict['AbsMag_Anexo'].append('+0,70')
this_dict['AbsMag_Anexo'].append('−0,29')
this_dict['AbsMag_Anexo'].append('15,49')
this_dict['AbsMag_Anexo'].append('15,49')
this_dict['AbsMag_Anexo'].append('-3,83')
this_dict['AbsMag_Anexo'].append('-0,88 ')
this_dict['AbsMag_Anexo'].append('-2,84')
this_dict['AbsMag_Anexo'].append('-5,28')
this_dict['AbsMag_Anexo'].append('-3,62')
this_dict['AbsMag_Anexo'].append('+0,78')
this_dict['AbsMag_Anexo'].append('-5,05')
this_dict['AbsMag_Anexo'].append('+1,30')
this_dict['AbsMag_Anexo'].append('-2,75')
this_dict['AbsMag_Anexo'].append('-1,45')
this_dict['AbsMag_Anexo'].append('+0,94 ')
this_dict['AbsMag_Anexo'].append('0.58')
this_dict['AbsMag_Anexo'].append('+13,08')
this_dict['AbsMag_Anexo'].append('-2,14')
this_dict['AbsMag_Anexo'].append('-3,03')
this_dict['AbsMag_Anexo'].append('2.22')
this_dict['AbsMag_Anexo'].append('-1,81')
this_dict['AbsMag_Anexo'].append('-7,2 (aprox)')
this_dict['AbsMag_Anexo'].append('-1,66')
this_dict['AbsMag_Anexo'].append('-0,74')
this_dict['AbsMag_Anexo'].append('+4,65')
this_dict['AbsMag_Anexo'].append('-1,52')
this_dict['AbsMag_Anexo'].append('+1,73')

In [ ]:
len(this_dict['AbsMag_Anexo'])

In [ ]:
tb = tb.with_columns(
    'Vmag', this_dict['Vmag'],
    'AbsMag (Eric)', this_dict['AbsMag'],
    'Abs Mag (ANEXO)', this_dict['AbsMag_Anexo']
)
tb

In [ ]:
#tb.to_csv('AnexoEstrellasProminentes2.csv')

In [ ]:
c = Table.read_table('AnexoEstrellasProminentes2.csv')
c

# HIP 1 ~ 10000

####    677: α Andromedae (Alpheratz o Sirrah) | Distancia 	97 ± 2 años-luz | Magnitud visual 	+2,02 a +2,06 | Magnitud absoluta 	-0,30 

In [ ]:
hip_main.where(0, are.equal_to(677)).select(0, 6, 3, 5)

####    3419: β Ceti (Deneb Kaitos o Diphda) | Distancia 	96 ± 2 años luz | Magnitud visual 	+2,04 | Magnitud absoluta 	-0,31 

In [ ]:
hip_main.where(0, are.equal_to(3419)).select(0, 6, 3, 5)

####    3829: Estrella de Van Maanen | Distancia 	14,1 ± 0,1 años luz | Magnitud visual 	+12,4 | Magnitud absoluta 	+14,2 

In [ ]:
hip_main.where(0, are.equal_to(3829)).select(0, 6, 3, 5)

####    5447: β Andromedae (Mirach) | Distancia 	199 ± 9 años luz | Magnitud visual 	+2,01 a +2,10 | Magnitud absoluta 	-1,87 

In [ ]:
hip_main.where(0, are.equal_to(5447)).select(0, 6, 3, 5)

####    7588: α Eridani (Achernar) | Distancia 144 años luz | Mag. aparente (V)	+0.45 (var) | Magnitud absoluta	-2.77

In [ ]:
hip_main.where(0, are.equal_to(7588)).select(0, 6, 3, 5)

####    9884: α Arietis (Hamal) | Distancia 	66 ± 1 años luz | Magnitud visual 	+2,00 | Magnitud absoluta 	+0,48 

In [ ]:
hip_main.where(0, are.equal_to(9884)).select(0, 6, 3, 5)

# HIP 10001 ~ 20000

####    10826: ο Ceti (Mira) | Distancia 	418 años luz (aprox) | Magnitud visual 	+2,0 a +10,1 (variable) | Magnitud absoluta 	+0,93 

In [ ]:
hip_main.where(0, are.equal_to(10826)).select(0, 6, 3, 5)

####    11767: α Ursae Minoris (Polaris) | Distancia	431 años luz (132 pc) | Mag. aparente (V)	+1,97 | Magnitud absoluta	-3,65

In [22]:
hip_main.where(0, are.equal_to(11767)).select(0, 1, 2, 6, 3, 5)

HIP   | RAdeg   | DEdeg   | dist (pc) | Vmag | abs_mag
11767 | 37.9461 | 89.2641 | 132.275   | 1.97 | -3.63739

In [23]:
pcToLy(132.275)

431.2165

####    15863: α Persei (Mirfak o Algenib) | Distancia 	590 años luz (aprox) | Magnitud visual 	+1,79 | Magnitud absoluta 	-4,50 

In [ ]:
hip_main.where(0, are.equal_to(15863)).select(0, 6, 3, 5)

# HIP 20001 ~ 30000

####    21421: α Tauri (Aldebarán) | Distancia	65,1 ± 1 años luz | Mag. aparente (V)	+0,85 / +13,50 | Magnitud absoluta	−0,63 / +11,98

In [ ]:
hip_main.where(0, are.equal_to(21421)).select(0, 6, 3, 5)

####    24186: Estrella de Kapteyn | Distancia 	12,8 ± 0,04 años luz | Magnitud visual 	+8,89 | Magnitud absoluta 	+10,88 

In [ ]:
hip_main.where(0, are.equal_to(24186)).select(0, 6, 3, 5)

####    24436: β Orionis (Rigel) | Distancia 	860 años luz (aprox) | Magnitud visual 	+0,18 | Magnitud absoluta 	-6,7 

In [ ]:
hip_main.where(0, are.equal_to(24436)).select(0, 6, 3, 5)

####    24608: α Aurigae (Capella) | Distancia 	42,2 ± 0,5 años luz | Magnitud visual 	+0,71 / +0,96 | Magnitud absoluta 	-0,5 (+0,14 / +0,29)

In [ ]:
hip_main.where(0, are.equal_to(24608)).select(0, 6, 3, 5)

####    25336: γ Orionis (Bellatrix) | Distancia 	240 ± 20 años luz | Magnitud visual 	+1,64 | Magnitud absoluta 	-2,72 

In [ ]:
hip_main.where(0, are.equal_to(25336)).select(0, 6, 3, 5)

####    25428: β Tauri (Elnath) | Distancia 	131 ± 5 años luz | Magnitud visual 	+1,68 | Magnitud absoluta 	-1,34 

In [ ]:
hip_main.where(0, are.equal_to(25428)).select(0, 6, 3, 5)

####    25930: δ Orionis (Mintaka) | Distancia	915 años luz | Mag. aparente (V)	+2,23 (conjunta) | Magnitud absoluta	-4,99 (conjunta)

In [ ]:
hip_main.where(0, are.equal_to(25930)).select(0, 6, 3, 5)

####    26311: ε Orionis (Alnilam) | Distancia	1340 años luz (600 pc) | Mag. aparente (V)	+1.70 (1.64—1.74) | Magnitud absoluta	-6.95

In [ ]:
hip_main.where(0, are.equal_to(26311)).select(0, 6, 3, 5)

####    26727: ζ Orionis (Alnitak) | Distancia	700 años luz (230 pc) | Mag. aparente (V)	+2,03 / +4 / +4,2 | Magnitud absoluta	-5,10

In [ ]:
hip_main.where(0, are.equal_to(26727)).select(0, 6, 3, 5)

####    27366: κ Orionis (Saiph) | Distancia 	722 años luz | Magnitud visual 	+2,05 | Magnitud absoluta 	-4,68 

In [ ]:
hip_main.where(0, are.equal_to(27366)).select(0, 6, 3, 5)

####    27989: α Orionis (Betelgeuse) | Distancia	643 ± 146 años luz (197 ± 45 pc) | Mag. aparente (V)	0,42 (0,3 a 1,2) | Magnitud absoluta	-6,0 a -5,5

In [ ]:
hip_main.where(0, are.equal_to(27989)).select(0, 6, 3, 5)

####    28360: β Aurigae (Menkalinan) | Distancia 	82,1 años luz | Magnitud visual 	+1,90 (+2,56 / +2,80 / +14,1) | Magnitud absoluta 	-0,11 (conjunta) 

In [ ]:
hip_main.where(0, are.equal_to(28360)).select(0, 6, 3, 5)

# HIP 30001 ~ 40000

####    30324: β Canis Majoris (Mirzam o Murzim) | Distancia 	500 años luz | Magnitud visual 	+1,98 | Magnitud absoluta 	-3,95 

In [ ]:
hip_main.where(0, are.equal_to(30324)).select(0, 6, 3, 5)

####    30438: α Carinae (Canopo) | Distancia	309 años luz (94.7 pc) | Mag. aparente (V)	-0,72 | Magnitud absoluta	-5,53

In [ ]:
hip_main.where(0, are.equal_to(30438)).select(0, 6, 3, 5)

####    31681: γ Geminorum (Alhena) | Distancia 	105 ± 8 años luz | Magnitud visual 	+1,93 | Magnitud absoluta 	-0,61 

In [ ]:
hip_main.where(0, are.equal_to(31681)).select(0, 6, 3, 5)

####    32349: α Canis Majoris (Sirio) | Distancia	8,6 ± 0,04 años luz (2,64 ± 0,01 pc) | Mag. aparente (V)	−1,46 (A) / 8,30 (B) | Magnitud absoluta	1,42 (A) / 11,18 (B)

In [ ]:
hip_main.where(0, are.equal_to(32349)).select(0, 6, 3, 5)

####    33579: ε Canis Majoris (Adhara) | Distancia 	405 años luz | Magnitud visual 	+1,51 | Magnitud absoluta 	-4,10 

In [ ]:
hip_main.where(0, are.equal_to(33579)).select(0, 6, 3, 5)

####    34444: δ Canis Majoris (Wezen) | Distancia 	1800 años luz (aprox) | Magnitud visual 	+1,83 | Magnitud absoluta 	-6,87 

In [ ]:
hip_main.where(0, are.equal_to(34444)).select(0, 6, 3, 5)

####    36850: α Geminorum (Cástor) | Distancia 	51 ± 0,8 años luz | Magnitud visual 	+1,96 / +2,91 | Magnitud absoluta 	+1,22 / +2,8 

In [ ]:
hip_main.where(0, are.equal_to(36850)).select(0, 6, 3, 5)

####    37279: α Canis Minoris (Procyon) | Distancia	11,41 ± 0,03 años luz | Mag. aparente (V)	+0,34 | Magnitud absoluta	+2,65 / +13,04

In [ ]:
hip_main.where(0, are.equal_to(37279)).select(0, 6, 3, 5)

####    37826: β Geminorum (Pólux) | Distancia 	33,7 ± 0,3 años luz | Magnitud visual 	+1,15 | Magnitud absoluta 	+1,09 

In [ ]:
hip_main.where(0, are.equal_to(37826)).select(0, 6, 3, 5)

####    39953: γ Velorum (Suhail o Regor) | Distancia 	800 años luz (aprox.) | Magnitud visual 	+1,78 / +4,27 | Magnitud absoluta 	-5,31 / -2,79 

In [ ]:
hip_main.where(0, are.equal_to(39953)).select(0, 6, 3, 5)

# HIP 40001 ~ 50000

####    41037: ε Carinae (Avior) | Distancia 	630 ± 60 años luz | Magnitud visual 	+1,95 (+2,4 / +3,1) | Magnitud absoluta 	-4,58 (conjunta) 

In [ ]:
hip_main.where(0, are.equal_to(41037)).select(0, 6, 3, 5)

####    42913: δ Velorum | Distancia 	79,7 ± 0,7 años luz | Magnitud visual 	1,95 (2,3 / 3,4 / 5,1) | Magnitud absoluta 	0,4 / 1,5 / 3,6 

In [ ]:
hip_main.where(0, are.equal_to(42913)).select(0, 6, 3, 5)

####    45238: β Carinae (Miaplacidus) | Distancia 	111 ± 2 años luz | Magnitud visual 	+1,67 | Magnitud absoluta 	-1,00 

In [ ]:
hip_main.where(0, are.equal_to(45238)).select(0, 6, 3, 5)

####    46390: α Hydrae (Alphard) | Distancia 	177 ± 8 años luz | Magnitud visual 	+1,99 | Magnitud absoluta 	-1,69 

In [ ]:
hip_main.where(0, are.equal_to(46390)).select(0, 6, 3, 5)

####    49669: α Leonis (Regulus) | Distancia 	77 ± 1 años luz | Magnitud visual 	+1,35 / +8,14 / +13,5 | Magnitud absoluta 	-0,5 / +4,2 / +9,5 

In [ ]:
hip_main.where(0, are.equal_to(49669)).select(0, 6, 3, 5)

# HIP 50001 ~ 60000

####    50583: γ Leonis (Algieba) | Distancia 	131 ± 3 años luz | Magnitud visual 	+2,22(+2,61 / +3,8) | Magnitud absoluta 	-0,92 (conjunta) 

In [ ]:
hip_main.where(0, are.equal_to(50583)).select(0, 6, 3, 5)

####    54035: Lalande 21185 | Distancia 	8,31 ± 0,01 años luz | Magnitud visual 	+7,47 | Magnitud absoluta 	+10,46 

In [ ]:
hip_main.where(0, are.equal_to(54035)).select(0, 6, 3, 5)

####    54061: α Ursae Majoris (Dubhe) | Distancia 	124 ± 2 años luz | Magnitud visual 	+1,79 (+1,87 / +4,81 / +7,12) | Magnitud absoluta 	-1,09 / +1,83 / +4,22 

In [ ]:
hip_main.where(0, are.equal_to(54061)).select(0, 6, 3, 5)

####    54872: δ Leonis (Duhr o Zosma) | Distancia 	57,7 ± 0,8 años luz | Magnitud visual 	+2,56 | Magnitud absoluta 	+1,29 

In [ ]:
hip_main.where(0, are.equal_to(54872)).select(0, 6, 3, 5)

####    54879: θ Leonis (Chertan) | Distancia 	165 años luz | Magnitud visual 	+3,33 | Magnitud absoluta 	-0,35 

In [ ]:
hip_main.where(0, are.equal_to(54879)).select(0, 6, 3, 5)

####    56997: 61 Ursae Majoris | Distancia 	31,1 ± 0,2 años luz | Magnitud visual 	+5,31 | Magnitud absoluta 	+5,41 

In [ ]:
hip_main.where(0, are.equal_to(56997)).select(0, 6, 3, 5)

# Esta no coincide

####    57362: β Leonis (Denébola) | Distancia 	35,9 ± 0,2 años luz | Magnitud visual 	+2,14 | Magnitud absoluta 	+1,91 

In [ ]:
hip_main.where(0, are.equal_to(57362)).select(0, 6, 3, 5)

####    57757: β Virginis (Zavijava) | Distancia 	35,6 ± 0,3 años luz | Magnitud visual 	+3,61 | Magnitud absoluta 	+3,40 

In [ ]:
hip_main.where(0, are.equal_to(57757)).select(0, 6, 3, 5)

####    58001: γ Ursae Majoris (Phecda) | Distancia 	84 ± 1 años luz | Magnitud visual 	+2,44 | Magnitud absoluta 	+0,36 

In [ ]:
hip_main.where(0, are.equal_to(58001)).select(0, 6, 3, 5)

####    59196: δ Centauri | Distancia 	395 años luz | Magnitud visual 	+2,58 | Magnitud absoluta 	-2,82 

In [ ]:
hip_main.where(0, are.equal_to(59196)).select(0, 6, 3, 5)

# HIP 60001 ~ 70000

####    60351: 12 Comae Berenices | Distancia	273,394 años luz (83.823 pc) | Mag. aparente (V)	+4,78 | Magnitud absoluta	0,16

In [ ]:
hip_main.where(0, are.equal_to(60351)).select(0, 6, 3, 5)

####    60718: α Crucis (Acrux) | Distancia 	325 años luz | Magnitud visual 	0,77 (conjunta) | Magnitud absoluta 	-3,6 / -2,88 

In [ ]:
hip_main.where(0, are.equal_to(60718)).select(0, 6, 3, 5)

####    61084: γ Crucis (Gacrux) | Distancia 	88 ± 2 años luz | Magnitud visual 	+1,63 | Magnitud absoluta 	-0,56 

In [ ]:
hip_main.where(0, are.equal_to(61084)).select(0, 6, 3, 5)

####    62434: β Crucis (Becrux o Mimosa) | Distancia 	280 años luz | Magnitud visual 	+1,30 | Magnitud absoluta 	-3,92 

In [ ]:
hip_main.where(0, are.equal_to(62434)).select(0, 6, 3, 5)

####    62956: ε Ursae Majoris (Alioth) | Distancia 	81 ± 1 años luz | Magnitud visual 	+1,76 | Magnitud absoluta 	-0,22 

In [ ]:
hip_main.where(0, are.equal_to(62956)).select(0, 6, 3, 5)

####    63608: ε Virginis (Vindemiatrix) | Distancia	110 ± 1 años luz (33,6 pc) | Mag. aparente (V)	+2,83 | Magnitud absoluta	+0,22

In [ ]:
hip_main.where(0, are.equal_to(63608)).select(0, 6, 3, 5)

####    65474: α Virginis (Espiga) | Distancia 	260 ± 20 años luz | Magnitud visual 	+1,04 (conjunta) | Magnitud absoluta 	-3,55 (conjunta) 

In [ ]:
hip_main.where(0, are.equal_to(65474)).select(0, 6, 3, 5)

####    65835: R Hydrae | Distancia 	425 años luz (aprox) | Magnitud visual 	+6,40 | Magnitud absoluta 	-2,55 

In [ ]:
hip_main.where(0, are.equal_to(65835)).select(0, 6, 3, 5)

####    66249: ζ Virginis (Heze) | Distancia 	73,2 ± 1,5 años luz | Magnitud visual 	+3,371​ | Magnitud absoluta 	+1,62 

In [ ]:
hip_main.where(0, are.equal_to(66249)).select(0, 6, 3, 5)

####    67301: η Ursae Majoris (Alkaid o Benetnasch) | Distancia 	101 ± 2 años luz | Magnitud visual 	+1,85 | Magnitud absoluta 	-0,60 

In [ ]:
hip_main.where(0, are.equal_to(67301)).select(0, 6, 3, 5)

####    68702: β Centauri (Hadar o Agena) | Distancia 	352 años luz | Magnitud visual 	+0,60 | Magnitud absoluta 	-5,42 

In [ ]:
hip_main.where(0, are.equal_to(68702)).select(0, 6, 3, 5)

####    68933: θ Centauri (Menkent) | Distancia 	60,9 ± 0,9 años luz | Magnitud visual 	+2,06 | Magnitud absoluta 	+0,70 

In [ ]:
hip_main.where(0, are.equal_to(68933)).select(0, 6, 3, 5)

####    69673: α Bootis (Arturo) | Distancia	36,7 ± 0,3 años luz | Mag. aparente (V)	−0,04 | Magnitud absoluta	−0,29

In [ ]:
hip_main.where(0, are.equal_to(69673)).select(0, 6, 3, 5)

# HIP 70001 ~ 80000

####    71681: α Centauri B (Rigil Kentaurus o Toliman) | Distancia 	4.5 años luz | Magnitud visual 	-0,01 | Magnitud absoluta 	15,49 

In [ ]:
hip_main.where(0, are.equal_to(71681)).select(0, 6, 3, 5)

# Estas dos no coinciden

####    71683: α Centauri A (Rigil Kentaurus o Toliman) | Distancia 	4.5 años luz | Magnitud visual 	-0,01 | Magnitud absoluta 	15,49 

In [ ]:
hip_main.where(0, are.equal_to(71683)).select(0, 6, 3, 5)

####    71860: α Lupi (Kakkab o Men) | Distancia 	550 años luz | Magnitud visual 	+2,30 | Magnitud absoluta 	-3,83 

In [ ]:
hip_main.where(0, are.equal_to(71860)).select(0, 6, 3, 5)

####    72607: β Ursae Minoris (Kochab) | Distancia	129 años luz (39 pc) | Mag. aparente (V)	2.07 | Magnitud absoluta	-0,88

In [ ]:
hip_main.where(0, are.equal_to(72607)).select(0, 6, 3, 5)

####    75097: γ Ursae Minoris (Pherkad) | Distancia 	480 ± 30 años luz | Magnitud visual 	+3,00 | Magnitud absoluta 	-2,84 

In [ ]:
hip_main.where(0, are.equal_to(75097)).select(0, 6, 3, 5)

# HIP 80001 ~ 90000

####    80763: α Scorpii (Antares) | Distancia	550 años luz | Mag. aparente (V)	+1,091​| Magnitud absoluta	-5,28

In [ ]:
hip_main.where(0, are.equal_to(80763)).select(0, 6, 3, 5)

####    82273: α Trianguli Australis (Atria) | Distancia 	415 ± 30 años luz | Magnitud visual 	+1,91 | Magnitud absoluta 	-3,62 

In [ ]:
hip_main.where(0, are.equal_to(82273)).select(0, 6, 3, 5)

####    82396: ε Scorpii (Wei) | Distancia 	65 ± 1 años luz | Magnitud visual 	+2,29 | Magnitud absoluta 	+0,78 

In [ ]:
hip_main.where(0, are.equal_to(82396)).select(0, 6, 3, 5)

####    85927: λ Scorpii (Shaula) | Distancia 	570 años luz ? | Magnitud visual 	+1,62 | Magnitud absoluta 	-5,05 

In [ ]:
hip_main.where(0, are.equal_to(85927)).select(0, 6, 3, 5)

####    86032: α Ophiuchi (Rasalhague o Ras Alhague) | Distancia 	46,7 ± 0,6 años luz | Magnitud visual 	+2,08 | Magnitud absoluta 	+1,30 

In [ ]:
hip_main.where(0, are.equal_to(86032)).select(0, 6, 3, 5)

####    86228: θ Scorpii (Sargas o Girtab) | Distancia 	270 ± 20 años luz | Magnitud visual 	+1,86 | Magnitud absoluta 	-2,75 

In [ ]:
hip_main.where(0, are.equal_to(86228)).select(0, 6, 3, 5)

# HIP 90001 ~ 10000

####    90185: ε Sagittarii (Kaus Australis) | Distancia 	145 ± 7 años luz | Magnitud visual 	+1,79 | Magnitud absoluta 	-1,45 

In [ ]:
hip_main.where(0, are.equal_to(90185)).select(0, 6, 3, 5)

####    90496: λ Sagittarii (Kaus Borealis) | Distancia 	77,3 ± 1,6 años luz | Magnitud visual 	+2,81 | Magnitud absoluta 	+0,94 

In [ ]:
hip_main.where(0, are.equal_to(90496)).select(0, 6, 3, 5)

####    91262: α Lyrae (Vega) | Distancia	25.3 ± 0.1 años luz (7.76 ± 0.03 pc) | Mag. aparente (V)	0.03 | Magnitud absoluta	0.58

In [ ]:
hip_main.where(0, are.equal_to(91262)).select(0, 6, 3, 5)

####    92403: Ross 154 | Distancia 	9,68 ± 0,05 años luz | Magnitud visual 	+10,44 | Magnitud absoluta 	+13,08 

In [ ]:
hip_main.where(0, are.equal_to(92403)).select(0, 6, 3, 5)

####    92855: σ Sagittarii (Nunki) | Distancia	224 ± 14 años luz (69 ± 4 pc) | Mag. aparente (V)	+2,05 | Magnitud absoluta	-2,14

In [ ]:
hip_main.where(0, are.equal_to(92855)).select(0, 6, 3, 5)

####    97278: γ Aquilae (Tarazed) | Distancia 	461 ± 49 años luz | Magnitud visual 	+2,72 | Magnitud absoluta 	-3,03 

In [ ]:
hip_main.where(0, are.equal_to(97278)).select(0, 6, 3, 5)

####    97649: α Aquilae (Altair) | Distancia	16.73 ± 5.05 años luz (5.13 ± 0.01 pc) | Mag. aparente (V)	+0.76 | Magnitud absoluta	2.22

In [ ]:
hip_main.where(0, are.equal_to(97649)).select(0, 6, 3, 5)

# HIP 100001 ~ 110000

####    100751: α Pavonis (Peacock) | Distancia 	183 ± 7 años luz | Magnitud visual 	+1,91 | Magnitud absoluta 	-1,81 

In [ ]:
hip_main.where(0, are.equal_to(100751)).select(0, 6, 3, 5)

####    102098: α Cygni (Deneb, Arided o Aridif) | Distancia 	1425 años luz (aprox) | Magnitud visual 	+1,25 | Magnitud absoluta 	-7,2 (aprox) 

In [ ]:
hip_main.where(0, are.equal_to(102098)).select(0, 6, 3, 5)

####    105881: ζ Capricorni (Marakk) | Distancia 	398 ± 44 años luz | Magnitud visual 	+3,77 | Magnitud absoluta 	-1,66 

In [ ]:
hip_main.where(0, are.equal_to(105881)).select(0, 6, 3, 5)

####    109268: α Gruis (Alnair) | Distancia 	101 ± 3 años luz | Magnitud visual 	+1,73 | Magnitud absoluta 	-0,74 

In [ ]:
hip_main.where(0, are.equal_to(109268)).select(0, 6, 3, 5)

# Esta no coincide

####    109358: β Canum Venaticorum (Asterion o Chara) | Distancia 	27,4 ± 0,2 años luz | Magnitud visual 	+4,26 | Magnitud absoluta 	+4,65 

In [ ]:
hip_main.where(0, are.equal_to(109358)).select(0, 6, 3, 5)

# HIP 110001 ~ 120000

####    112122: β Gruis (Gruid) | Distancia 	170 ± 7 años luz | Magnitud visual 	+2,0 a +2,3 | Magnitud absoluta 	-1,52 

In [ ]:
hip_main.where(0, are.equal_to(112122)).select(0, 6, 3, 5)

####    113368: α Piscis Austrini (Fomalhaut) | Distancia 	25 ± 0,1 años luz | Magnitud visual 	+1,16 | Magnitud absoluta 	+1,73 

In [ ]:
hip_main.where(0, are.equal_to(113368)).select(0, 6, 3, 5)